### PR TITLE
feat: add category and product entities

### PR DESCRIPTION
**Pull request by CarlosDanielS3**
add category and product entities and their basic routes

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -11,13 +11,36 @@ datasource db {
...
   @@map("users")
+}
+
+model Category {
+  id         Int       @id @default(autoincrement())
+  name       String
+  father_id  Int?
+  created_at DateTime  @default(now())
+  updated_at DateTime  @updatedAt
+  Product    Product[]
```
Comment (ArthurRF) [inline prisma/schema.prisma]: When there's a one-to-many relation on prisma couldn't we use plural to make it easier to understand? like:
```suggestion
  products   Product[]
```

File: src/infrastructure/repositories/catogorie.repositorie.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { PrismaService } from '../prisma/prisma.service';
+
+import { ICategoryRepository } from '@/domain/repositories/category.repository';
+import {
+  UpdateCategoryDto,
+  AddCategoryDto,
+} from '@/presentation/category/category.dto';
+
+@Injectable()
+export class DatabaseCategoryRepository implements ICategoryRepository {
```
Comment (ArthurRF) [inline src/infrastructure/repositories/catogorie.repositorie.ts]: The name of the file is `categorie-repositorie`, it must be category-repository, just to keep the pattern

File: src/infrastructure/repositories/catogorie.repositorie.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    });
+    return categoryEntities;
+  }
+  async findById(id: number): Promise<Category> {
+    return await this.prisma.category.findUniqueOrThrow({
+      where: { id },
+    });
+  }
+  async deleteById(id: number): Promise<void> {
+    await this.prisma.product.findUniqueOrThrow({
```
Comment (ArthurRF) [inline src/infrastructure/repositories/catogorie.repositorie.ts]:
```suggestion
    await this.prisma.category.findUniqueOrThrow({
```

File: src/infrastructure/repositories/user.repository.ts [hunk 4]
```diff
@@ -30,7 +30,10 @@ export class DatabaseUserRepository implements IUserRepository {
     });
   }
   async deleteById(id: number): Promise<void> {
-    await this.prisma.user.deleteMany({ where: { id } });
+    await this.prisma.product.findUniqueOrThrow({
```
Comment (ArthurRF) [inline src/infrastructure/repositories/user.repository.ts]:
```suggestion
    await this.prisma.user.findUniqueOrThrow({
```

File: src/presentation/product/product.controller.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  async update(
+    @Param('id') id: string,
+    @Body() body: UpdateProductDto,
+  ): Promise<void> {
+    return await this.productUseCase.update(Number(id), body);
+  }
+
+  @Delete('/:id')
+  @HttpCode(204)
+  async deleteUser(@Param('id') id: string): Promise<void> {
```
Comment (ArthurRF) [inline src/presentation/product/product.controller.ts]:
```suggestion
  async deleteProduct(@Param('id') id: string): Promise<void> {
```

File: src/presentation/category/category.controller.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  async update(
+    @Param('id') id: string,
+    @Body() body: UpdateCategoryDto,
+  ): Promise<void> {
+    return await this.categoryUseCase.update(Number(id), body);
+  }
+
+  @Delete('/:id')
+  @HttpCode(204)
+  async deleteUser(@Param('id') id: string): Promise<void> {
```
Comment (ArthurRF) [inline src/presentation/category/category.controller.ts]:
```suggestion
  async deleteCategory(@Param('id') id: string): Promise<void> {
```

**Review by ArthurRF — CHANGES_REQUESTED** [review]

**Comment by CarlosDanielS3** [comment]
Fixed pr issues

**Review by ArthurRF — APPROVED** [review]

**Pull request merged**